blaze/server/spider.py

```diff
@@ -72,28 +73,47 @@ def data_spider(path,
     """
     # NOTE: this is named `data_spider` rather than just `spider` to
     # disambiguate this function from the `blaze.server.spider` module.
-    return {
-        os.path.basename(path): _spider(path, ignore=ignore,
-                                        followlinks=followlinks,
-                                        hidden=hidden,
-                                        extra_kwargs=extra_kwargs)
-    }
+    return {os.path.basename(path): _spider(path,
+                                            ignore=ignore,
+                                            followlinks=followlinks,
+                                            hidden=hidden,
+                                            extra_kwargs=extra_kwargs)}
 
 
-def from_yaml(path, ignore=(ValueError, NotImplementedError), followlinks=True,
-              hidden=False):
+@contextmanager
+def manage_chdir(root):
+    """Context manager that changes to ``root`` directory on ``__entry__`` and
```

Comment:
this data model method here is `__enter__`; however, I think it is fine to just say on enter or on exit (without the underscores)